h2integrate/postprocess/mapping.py

```diff
@@ -0,0 +1,852 @@
+    )
+
+    # format tick marks on colorbar
+    inset_ax.tick_params(
+        direction=map_preferences.colorbar_tick_direction,
+        labelsize=map_preferences.colorbar_tick_label_font_size,
+    )
+
+    # format color bar legend offset text and position (exp notation for values if applicable)
+    # By default, this is set such that if the values of the colorbar legend are smaller or larger
+    # than -3,4 decimal places (0.001-9999) then use exponential notation
+    cbar.formatter.set_scientific(map_preferences.colorbar_tick_label_use_exp_notation)
+    cbar.formatter.set_powerlimits(map_preferences.colorbar_tick_label_exp_notation_decimal_limit)
+
+    offset_text = cbar.ax.xaxis.get_offset_text()
+    offset_text.set_fontsize(map_preferences.colorbar_tick_label_font_size)
+
+    # Dynamically set the exponential notation text x position based on the colorbar's width
+    # Defaults to +2.5% relative to the colorbar width, normalized to the inset axis of the colorbar
+    # Ex: colorbar_width = 20%, dynamic offset = (22.5/20) = 1.125
+    # NOTE: hardcoding this for ease of handling and reducing inputs, can be changed if needed
+    colorbar_width = float(map_preferences.colorbar_width[:-1])
+    dyn_exp_notation_x_offset = (colorbar_width + 2.5) / colorbar_width
+    offset_text.set_position((dyn_exp_notation_x_offset, np.nan))
+    # Set the expontential notation y position centered on the colorbar
+    # NOTE: hardcoding this for ease of handling and reducing inputs, can be changed if needed
+    cbar.ax.xaxis.OFFSETTEXTPAD = -24
+
+    # Calculate appropriate bounds for map based on coordinates of data used in plots
+    coord_range_dict = calculate_geodataframe_total_bounds(*gdfs_for_bounds)
+
+    # Set basemap extent based on calculated bounds and padding fractions
+    left_pad = coord_range_dict["x_range"] * map_preferences.basemap_leftpad
+    right_pad = coord_range_dict["x_range"] * map_preferences.basemap_rightpad
+    upper_pad = coord_range_dict["y_range"] * map_preferences.basemap_upperpad
+    lower_pad = coord_range_dict["y_range"] * map_preferences.basemap_lowerpad
+
+    ax.set_xlim(coord_range_dict["min_x"] - left_pad, coord_range_dict["max_x"] + right_pad)
+    ax.set_ylim(coord_range_dict["min_y"] - lower_pad, coord_range_dict["max_y"] + upper_pad)
+    ax.set_axis_off()
+    ax.set_title(map_preferences.figure_title)
+
+    # Plot basemap with contextily
+    ctx.add_basemap(
+        ax,
+        crs=map_preferences.web_map_crs,
+        source=map_preferences.basemap_provider,
+        zoom=map_preferences.basemap_zoom,
+    )
+
+    # Show the plot if show_plot == True
+    # NOTE: when plotting multiple layers, set this to True only when plotting the last layer
+    if show_plot:
+        plt.show()
+
+    # Save figure if save_plot_fpath is present
+    if save_plot_fpath is not None:
+        if isinstance(save_plot_fpath, str):
+            save_plot_fpath = Path(save_plot_fpath)
```

Comment:
`save_plot_fpath` doesn't need to be a `Path` object if it's only being passed to `savefig`.